guides/21_Background_Filter_VideoFX_Processor.md

```diff
@@ -0,0 +1,375 @@
+- [Example background filter](#example-bg-filter)
+
+# About using background filters<a id="about-using-background-filters"></a>
+
+## Using the cross\-origin opener policy<a id="cross-origin-policy"></a>
+
+To limit memory usage, the module prefers to use a `SharedArrayBuffer` for processing\. However, this requires that you carefully configure web security\. You must set both of these headers when serving your application HTML:
+
+```
+Cross-Origin-Opener-Policy: same-origin
+Cross-Origin-Embedder-Policy: require-corp
+```
+
+The server must set these because they have no meta\-tag equivalents\. If you don't set these headers, the background filters may use slightly more RAM\.
+
+Background filters can be CPU\-intensive and GPU\-intensive\. Some mobile devices and lower\-specification laptop or desktop computers may not have the power to run the filters along with multiple video streams\.
+
+## SIMD support<a id="simd-support"></a>
+
+Background filters are more efficient in environments that support Single Instruction, Multiple Data (SIMD). The filters use less CPU for a given complexity level when you enable SIMD\. Low\-powered devices running browsers without SIMD support may not run background filters\.
+
+## WebGL2 support<a id="webgl2-support"></a>
+
+The `VideoFxProcessor` object requires browsers that support WebGL2 in order to access the GPU on the client device\.
+
+## Content delivery and bandwidth<a id="delivery-caching-bandwidth"></a>
+
+An Amazon content delivery network loads the machine\-learning\-model files for background filters at runtime\. This provides low\-latency global distribution without the need to serve a full suite of files as part of your application\. However, loading model files can add latency to parts of your application\. To help mitigate that impact, browsers cache the model files indefinitely\. That cache makes subsequent loads significantly faster\. As a best practice, check for supported browsers, then create the background filter resources when users may not notice any latency\. For example, you can download model files while users wait in a lobby, or while they use a device picker\.
+
+Your application must connect to the following:
++ Amazon Chime SDK media services\. 
++ Amazon CloudFront via HTTPS (port 443).
+
+All requests are to subdomains of `sdkassets.chime.aws`\. Applications that can't access the content delivery network or don't include the correct domain in their [content security policy](#content-security) will fail their support checks and be unable to use the filters\.
+
+For more information about CloudFront’s IP address ranges, see [ Locations and IP address ranges of CloudFront edge servers](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/LocationsOfEdgeServers.html) in the *Amazon CloudFront Developer Guide*\.
+
+## Browser compatibility<a id="filters-browser-compat"></a>
+
+The following table lists the browsers and version that support background filters\.
```

Comment:
Should we clarify here that Android is currently not supported and if they need Android support please use the older ones?

Reply:
Asking my manager to confirm the change. WIll update after that.

Reply:
Fixed!